includes/services.inc.php

```diff
@@ -108,13 +109,66 @@ function delete_service($service = null)
 function discover_service($device, $service)
 {
     if (! dbFetchCell('SELECT COUNT(service_id) FROM `services` WHERE `service_type`= ? AND `device_id` = ?', [$service, $device['device_id']])) {
-        add_service($device, $service, "(Auto discovered) $service");
+        add_service($device, $service, "$service Monitoring (Auto Discovered)", null, null, 0, 0, 0, "AUTO: $service", 0);
         log_event('Autodiscovered service: type ' . mres($service), $device, 'service', 2);
         echo '+';
     }
     echo "$service ";
 }
 
+function delete_service_template($service_template = null)
+{
+    if (! is_numeric($service_template)) {
+        return false;
+    }
+    remove_service_template($service_template);
+
+    return dbDelete('service_templates', '`id` =  ?', [$service_template]);
+}
+
+function discover_service_template($service_template = null)
+{
+    $services_template = ServiceTemplate::getServiceTemplate($service_template);
+    $status = null;
+    foreach (DB::table('device_group_device')->where('device_group_id', $services_template[0]['device_group_id'])->pluck('device_id') as $device) {
+        foreach (DB::table('services')->where('service_template_id', $service_template)->where('device_id', $device)->where('service_template_changed', '!=', $services_template[0]['changed'])->pluck('service_id') as $service) {
+            $update = ['service_desc' => $services_template[0]['desc'], 'service_ip' => $services_template[0]['ip'], 'service_param' => $services_template[0]['param'], 'service_ignore' => $services_template[0]['ignore'], 'service_disabled' => $services_template[0]['disabled'], 'service_template_id' => $services_template[0]['id'], 'service_name' => $services_template[0]['name'], 'service_template_changed' => $services_template[0]['changed']];
+            edit_service($update, $service['service_id']);
+            log_event("Updated Service: {$services_template[0]['name']} from Service Template ID: {$services_template[0]['id']}", $device, 'service', 2);
+            $status = 1;
+        }
+        if (! DB::raw('COUNT(service_id)')->from('services')->where('service_template_id', $service_template)->where('device_id', $device)) {
```

Comment:
Service::where('service_template_id', $service_template)->where('device_id', $device)->count()

Reply:
replaced, thank you!